README.md

```diff
@@ -1,208 +1,262 @@
+For more information on contributing to this repository visit [Contributing to a GitHub repository](https://world.mendix.com/display/howto6/Contributing+to+a+GitHub+repository)!
+
+## Implementation Guide
+
+### Step 1 - Create an mpk of the PushNotifications module
+
+We need to extract the module from this project before starting with the implementation. Walk through these following steps:
+
+1. Clone this project or download it as ZIP and extract it.
+2. Open the `PushNotifications.mpr` which is located in the `test` directory in the root of the project with a Mendix Modeler.
+3. Right-click on the `PushNotifications` module (inside Project Explorer pane), select `Export module package...` and save the mpk file.
+
+<img src="assets/images/implementation guide/Preparation export Push Notification module.JPG"/>
+
+### Step 2 - Install module dependencies
+
+First, open your existing Mendix project (or create a new one). The Push Notification module has two dependencies: `CommunityCommons` and `Encryption` module. Include these two dependencies by downloading it from the AppStore.
+
+> Note: importing the Encryption module will trigger errors because it contains a reference to a non-existant layout. Fix it by assigning a master layout of `Encryption.ResponsiveLayout_Certificate` page to some other layout (in this specific use case it is not really important which layout is used).
+>
+> <img src="assets/images/implementation guide/Fix Encryption module.JPG"/>
+
+### Step 3 - Import the PushNotification module
+
+Import the created mpk file from Step 1 into your Mendix project. To do this, right-click on an empty space on the Project Explorer pane, select `Import module package...`, choose the mpk file, and add it as a new module.
+
+<img src="assets/images/implementation guide/Import Push Notification module.JPG"/>
+
+<img src="assets/images/implementation guide/Import Push Notification module as a new module.JPG"/>
+
+### Step 4 - Update component.json file
+
+Find the `theme\components.json` file in your project. This file contains (among others) the dependencies of the to-be-created mobile hybrid application. Update `theme\components.json` by adding `"widgets/pushNotifications/lib/PushNotification.js"` as an element of the `js` array so it would look like this:
+
 ```
-{
-    "files": {
-        "css": ["lib/bootstrap/css/bootstrap.min.css", "mxclientsystem/mxui/ui/mxui.css", "css/theme.css"],
-        "js": ["mxclientsystem/mxui/mxui.js", "js/jquerymin.js", "widgets/pushNotifications/lib/PushNotification.js"]
-    },
-    "cachebust": "{{cachebust}}"
+{  
+   "files":{  
+      "css":[  
+         "lib/bootstrap/css/bootstrap.min.css",
+         "mxclientsystem/mxui/ui/mxui.css",
+         "styles/css/lib/lib.css",
+         "styles/css/custom/custom.css"
+      ],
+      "js":[  
+         "mxclientsystem/mxui/mxui.js",
+         "widgets/pushNotifications/lib/PushNotification.js"
+      ]
+   },
+   "cachebust":"635689412670032000"
```

Comment:
get rid of the specific cachebust, so people don't start including that... see how it was templated in the old version.